pkg/ingester/stream.go

```diff
@@ -411,13 +411,14 @@ func (s *stream) validateEntries(ctx context.Context, entries []logproto.Entry,
 
 		lineBytes := len(entries[i].Line)
 		totalBytes += lineBytes
+		metadataBytes := util.StructuredMetadataSize(entries[i].StructuredMetadata)
 
 		now := time.Now()
 		if !rateLimitWholeStream && !s.limiter.AllowN(now, len(entries[i].Line)) {
```

Comment:
Same as before. I think we should account for structured metadata to assert the rate limits